dotcom-rendering/src/model/enhance-ad-placeholders.ts

```diff
@@ -65,13 +83,16 @@ const insertAdPlaceholders = (elements: FEElement[]): FEElement[] => {
 			currentElement: FEElement,
 			idx: number,
 		): ReducerAccumulator => {
-			const paragraphCounter = isParagraph(currentElement)
-				? prev.paragraphCounter + 1
-				: prev.paragraphCounter;
+			const blockCounter =
```

Comment:
I know it's not your addition but why only count paragraphs and now images too? And not embeds or whatever else

Reply:
I'm assuming for simplicity? Because we're looking at the server side page elements before they're rendered, we could check a long list of element types and basically end up with an apps version of spacefinder. Images seem sensible to include as there's so many of them! Perhaps we should have a think about what we want to support and where we draw the line to avoid the code getting super complicated?